packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/RealmObjectHelper.kt

```diff
@@ -759,6 +761,141 @@ internal object RealmObjectHelper {
+                                        createDetachedCopy(
+                                            mediator,
+                                            listObject,
+                                            currentDepth + 1,
+                                            maxDepth,
+                                            closeAfterCopy,
+                                            cache
+                                        )
+                                    )
+                                }
+                            }
+                            accessor.set(target, list)
+                        }
+                        else -> {
+                            throw IllegalStateException("Unknown type: ${property.type}")
+                        }
+                    }
+                }
+                CollectionType.RLM_COLLECTION_TYPE_SET -> {
+                    val elements: Set<Any?> = accessor.get(source) as Set<Any?>
+                    when (property.type) {
+                        PropertyType.RLM_PROPERTY_TYPE_INT,
+                        PropertyType.RLM_PROPERTY_TYPE_BOOL,
+                        PropertyType.RLM_PROPERTY_TYPE_STRING,
+                        PropertyType.RLM_PROPERTY_TYPE_BINARY,
+                        PropertyType.RLM_PROPERTY_TYPE_FLOAT,
+                        PropertyType.RLM_PROPERTY_TYPE_DOUBLE,
+                        PropertyType.RLM_PROPERTY_TYPE_TIMESTAMP,
+                        PropertyType.RLM_PROPERTY_TYPE_OBJECT_ID,
+                        PropertyType.RLM_PROPERTY_TYPE_UUID -> {
+                            accessor.set(target, elements.toRealmSet())
+                        }
+                        PropertyType.RLM_PROPERTY_TYPE_OBJECT -> {
+                            val set = UnmanagedRealmSet<BaseRealmObject>()
+                            if (currentDepth < maxDepth) {
+                                (elements as Set<BaseRealmObject>).forEach { realmObject: BaseRealmObject ->
+                                    set.add(
+                                        createDetachedCopy(
+                                            mediator,
+                                            realmObject,
+                                            currentDepth + 1,
+                                            maxDepth,
+                                            closeAfterCopy,
+                                            cache
+                                        )
+                                    )
+                                }
+                            }
+                            accessor.set(target, set)
+                        }
+                        else -> {
+                            throw IllegalStateException("Unknown type: ${property.type}")
+                        }
+                    }
+                }
+                else -> {
+                    TODO("Collection type ${property.collectionType} is not supported.")
```

Comment:
Still a todo or an `IllegalStateException` ?

Reply:
This should be an IllegalStateException